lychee-lib/src/extract/markdown.rs

```diff
@@ -1,35 +1,131 @@
-use pulldown_cmark::{Event as MDEvent, Parser, Tag};
+use pulldown_cmark::{Event, Parser, Tag};
 
 use crate::{extract::plaintext::extract_plaintext, types::raw_uri::RawUri};
 
 /// Extract unparsed URL strings from a Markdown string.
-pub(crate) fn extract_markdown(input: &str) -> Vec<RawUri> {
+pub(crate) fn extract_markdown(input: &str, skip_code_blocks: bool) -> Vec<RawUri> {
+    // In some cases it is undesirable to extract links from within code blocks,
+    // which is why we keep track of entries and exits while traversing the input.
+    let mut inside_code_block = false;
+
     let parser = Parser::new(input);
     parser
-        .flat_map(|event| match event {
-            MDEvent::Start(Tag::Link(_, uri, _)) => {
-                vec![RawUri {
+        .filter_map(|event| match event {
+            // A link. The first field is the link type, the second the destination URL and the third is a title.
+            Event::Start(Tag::Link(_, uri, _)) => {
+                Some(vec![RawUri {
                     text: uri.to_string(),
                     // Emulate `<a href="...">` tag here to be compatible with
                     // HTML links. We might consider using the actual Markdown
                     // `LinkType` for better granularity in the future
                     element: Some("a".to_string()),
                     attribute: Some("href".to_string()),
-                }]
+                }])
             }
-            MDEvent::Start(Tag::Image(_, uri, _)) => {
-                vec![RawUri {
+            // An image. The first field is the link type, the second the destination URL and the third is a title.
+            Event::Start(Tag::Image(_, uri, _)) => {
+                Some(vec![RawUri {
                     text: uri.to_string(),
                     // Emulate `<img src="...">` tag here to be compatible with
                     // HTML links. We might consider using the actual Markdown
                     // `LinkType` for better granularity in the future
                     element: Some("img".to_string()),
                     attribute: Some("src".to_string()),
-                }]
+                }])
+            }
+            // A code block (inline or fenced).
+            Event::Start(Tag::CodeBlock(_)) => {
+                inside_code_block = true;
+                None
+            }
+            Event::End(Tag::CodeBlock(_)) => {
+                inside_code_block = false;
+                None
+            }
+
```

Comment:
The code below is the main change: before we ignored the hint that we're inside a code block. Now we track that with `inside_code_block`.